thunder/executors/nvfuserex_impl.py

```diff
@@ -1953,7 +1953,11 @@ def where(
     nva = getnv(a, fd, lc_to_nv_map)
     nvb = getnv(b, fd, lc_to_nv_map)
 
-    return fd.ops.where(nvpred, nva, nvb)
+    # Determines result dtype
+    numbertype, tensordtype = utils.check_same_dtype(a, b)
+    dtype = tensordtype if tensordtype is not None else numbertype
```

Comment:
technically I think we only needed the cast when both inputs are scalars.

Reply:
This seems like a fine workaround, but should this eventually be fixed in nvfuser itself? Maybe we should get an issue to track that

Reply:
Just to clarify, the root cause is that nvfuser doesn't have `float` scalar input, but rather only allows `double`.

So not really a bug but rather a feature request. I'll open an issue to track that.

Reply:
haha -- however nvfuser wants to track the issue is fine

Reply:
https://github.com/NVIDIA/Fuser/issues/3816